### PR TITLE
Scripts Created

### PR DESCRIPTION
**Pull request by saadali5997**

### CODE REVIEW

File: task1.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
Comment (fahadsiddiqui) [inline task1.sh]: Remove the file `.task1.sh.swp` from git.

File: task2.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+echo "Creating task2 folder and creating 3 files in it"
+$(mkdir task2 && cd task2 && touch file1 && touch file2 && touch file3)
+echo "Creating task2_temp folder copying content of task2 in it"
+$(mkdir task2_temp && cp -a task2/. task2_temp/)
```
Comment (fahadsiddiqui) [inline task2.sh]: No need to put every command in a `$()`.

File: task2.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+echo "Creating task2 folder and creating 3 files in it"
+$(mkdir task2 && cd task2 && touch file1 && touch file2 && touch file3)
```
Comment (fahadsiddiqui) [inline task2.sh]: Avoid usage of `&&`. It adds complexity.
Make your script as easy as possible for future edits.

File: task2.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+echo "Creating task2_temp folder copying content of task2 in it"
+$(mkdir task2_temp && cp -a task2/. task2_temp/)
+
+ds=($(ls -d */))
+f1=${ds[0]:0:${#ds[0]}-1}
```
Comment (fahadsiddiqui) [inline task2.sh]: Explain these two lines with a comment.

File: task2.sh (hunk 2 above)
Comment (saadali5997) [reply task2.sh]: I am getting different results on same lines by removing "$( )", why could it be so?
For example, an error saying 
'task2/.': no such file or directory

Comment (saadali5997) [reply task2.sh]: Had to make slight changes on removing "$()".

File: task1.sh (hunk 1 above)
Comment (saadali5997) [reply task1.sh]: Done.

File: task2.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ds=($(ls -d */))
+
+#Getting the name of each variable/directory in the string form and removing "/" from the end so that names 
+#could be swapped
+
+f1=${ds[0]:0:${#ds[0]}-1}
+f2=${ds[1]:0:${#ds[1]}-1}
+
+echo "Swapping the names of those folders"
+$(mv $f1 tempo)
```
Comment (fahadsiddiqui) [inline task2.sh]: Remove `$()` from here too.

File: task1.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+#Change directory to folder we want to search in
+cd $2
+
+#Get the list of all directories, sub-directories & files in variable ds
+ds=$(ls -R)
```
Comment (fahadsiddiqui) [inline task1.sh]: Normally you name a variable with Capital letters in bash.

File: task2.sh [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+echo "Creating task2_temp folder, copying content of task2 in it"
+
+#Go back to parent directory of task2
+cd ../
+#Make a new directory task2_temp
+mkdir task2_temp
+#Copy contents of task2 to task2_temp
+cp -a task2/ task2_temp
+
+#Copying the names of all directories from pwd in array ds
```
Comment (fahadsiddiqui) [inline task2.sh]: Comments generally have the format
```
# i am a nice comment, see my first letter is small letter with a space after pound sign
```

File: task2.sh [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+ds=($(ls -d */))
+
+: 'Getting the name of each variable/directory in the string form from array  and removing "/" from the end so that names could be swapped'
```
Comment (fahadsiddiqui) [inline task2.sh]: What's that string for?

File: task2.sh [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+#get folder 1's name from index 0 and trim "/" from the end
+f1=${ds[0]:0:${#ds[0]}-1}
+
+#get folder 2's name from index 1 and trim "/" from the end
+f2=${ds[1]:0:${#ds[1]}-1}
+
+echo "Swapping the names of those folders"
```
Comment (fahadsiddiqui) [inline task2.sh]: Normally the progress strings end with `...` (ellipses) to make it look like "WAIT FOR THIS TO HAPPEN". Just a suggestion.

File: task2.sh [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+#Rename task2 directory to tempo
+mv $f1 tempo
+
+#Rename task2_temp to task2
+mv $f2 $f1
+
+#Rename tempo to task2_temp
+mv tempo $f2
```
Comment (fahadsiddiqui) [inline task2.sh]: Next time try giving very readable names to variables.

File: task2.sh (hunk 8 above)
Comment (saadali5997) [reply task2.sh]: For swapping names, we have to know the names. The string contains the name of folder, as variable.

**Pull request merged**